src/plugins/lwhpolar/README.md

```diff
@@ -0,0 +1,98 @@
+Polar Coordinate Positioning Plugin
+===================================
+
+Preface
+-------
+This plugin replaces the "rel" relative positioning plugin. 
+It includes and enhances its functionality.
+Using both will confuse the positioning, as the first one called replaces 
+relative positions by absolute ones, thus disabling the second one.
+
+Relative Positioning (Cartesian)
+--------------------------------
+The plugin uses the following attributes for a step:
+	data-rel-ref
+	data-rel-x
+	data-rel-y
+	data-rel-z
+
+If the attribute "data-rel-ref" is not given, relative coordinates will be measured
+from the position of the previous frame. If added, the step with the given id determines
+the origin. This allows to group steps to a topic without the risk of scattering them when 
+one of them is removed.
+In addition to plain numbers, which are pixel values, it is also possible to
+define relative positions as a multiple of normal step height and width, using
+a unit of "h" or "w", respectively, appended to the number.
+
+Example:
+	<div id="relstep" class="step" 
+		data-rel-ref="main_step" data-rel-x="1w" data-rel-y="0.5h" data-rel-z="0">
+			
+Relative coordinates not given are inherited from the previous step, so make sure
+to set them to zero if that behaviour is not desired.
+Absolute values always override relative values!
+
+Polar Positioning
+-----------------
+
+To place steps e.g. in a circle around a central thought, polar positioning is introduced here.
+It uses the following attributes for a step:
+	data-rel-ref
+	data-rho
+	data-phi
+	data-rel-rho
+	data-rel-phi
```

Comment:
OK, I'll look through them (which might affect the presentation I'm working on)

Reply:
Making polar coordinates always relative to something (either 0 or a certain frame) looks like a good idea. It might even eliminate the interference with the rel plugin.
My plugin still includes some organic growth, maybe it's time for the clippers...

Reply:
I tried around some to separate the two, however, if polar coordinates reference other frames, their positions have to be calculated first. If these are relative to a frame defined polar, that has to be known already. So there is no simple sequence - the two plugins would have a circular dependency.

Reply:
Ah I see. Now that makes sense.

Let's think about it... One solution would be that this plugin transforms polar coordinates into `rel`-coordinates. Then it could always run before `rel` plugin. Not architecturally nice to have such a chained dependency, just wanted to point out it's a solution that exists.

Reply:
That might work, but adds a dependency not only for the existence, but also for the sequence of plugins. On the other hand, it avoids duplicate code.
However, if someone changes the _rel_ plugin, this could break the _polar_ plugin as well.
So KISS beats DRY ;-)
For consistency (having absolute coordinates - either polar or cartesian - count from the origin and relative coordinates either from a given reference or incremental) I'm back to my original version with some changes and another angle for spherical coordinates. 
I'll test that and upload it soon for review.

Reply:
Thinking a bit more, it's quite possible that to do this "the right way", we need to refactor the positioning code further. impress.js creator @bartaz once shared his vision of a refactor where also the absolute `data-x/y/z` coordinates would be a plugin. And that plugin would be equal to `rel`, or `polar` or any other positioning plugin. The core impress.js would then contain only some 3D rendering code to move around in a more generic way, as instructed by whatever plugin is relevant for each step.

This might be the time to do that. (Would be after a 1.0.0 release.)

Reply:
Well, that would definitely be too late for my current presentation, so for now I have to stick to what I have...
But I'll be glad to help.

Reply:
Just an idea: If the loop over all steps would be running in the main program, calling the positioning plugins for each step, that one would be finally positioned before the next step is called, thus avoiding the problem of transient positions used as references.
But that would probably mix up the whole plugin structure.

Reply:
> one would be finally positioned before the next step is called

If we constrain `data-rel-ref` and `data-polar-ref` so that they can only reference a previous step, then that would work. That feels like a reasonable restriction IMO.

> But that would probably mix up the whole plugin structure.

While current pre-init plugins execute once on init, we would need a similar plugin that is called on init, but for each step in sequence. The step would be passed as argument, together with access to the internal data for all (previous) steps, and an object where a plugin can provide the computed position (or any other attribute) for the present step.

> Well, that would definitely be too late for my current presentation, so for now I have to stick to what I have...

Also would need to add position related testing before any refactorin. Several PR worth of work here.

> But I'll be glad to help.

Thanks! I started with the `rel` plugin myself. Step positioning is a good way to get to the core of impress.js!

Reply:
I never had the idea of referring to a later defined step with some kind of `data-rel-to` attribute. Such an implementation would create the possibility of defining circular and/or offending dependencies that are impossible to resolve such as 
```html
<div id="one" class="step" data-rel-x="1000" data-rel-y="1000" data-rel-to="three"></div>
<div id="two" class="step" data-rel-x="1000" data-rel-y="1000" data-rel-to="one"></div>
<div id="three" class="step" data-rel-x="1000" data-rel-y="1000" data-rel-to="two"></div>
```
or even simpler
```html
<div id="one" class="step" data-rel-x="1000" data-rel-y="1000" data-rel-to="one"></div>
```
If someone really needs to refer to a step that is shown later, that can easily be achieved with the help of the `goto` plugin. A hint/example in the documentation would be a help for users.

I would strongly vote to forbid any way of referencing later defined steps. Even if one can detect such circles it makes the code much more complicated as well as much more slower.

Reply:
It's a good point. Let's say it is forbidden.

Reply:
I agree. It would never work properly.

I have reworked the plugin to support several separate positioning substeps (absolute, relative, polar etc.), which might be used for a new, stepwise plugin approach. Please have a look at pull request #678.
Not really simple, but works fine for me.

Reply:
@henrikingo If you are reworking the `rel` plugin, what about adding a `data-rel-rotate` attribute to make steps rotate stepwise? That way one could easily make steps e.g. point towards the center of a circle without calculating all those angles by hand.

Reply:
First of all, just to clarify, we will not be reworking the rel plugin. (From this PR, we may add the idea about a data-rel-ref attribute to make the positioning relative to some other step than the immediately previous one.)

As for polar/spherical plugin, I have actually considered the idea of relative rotation as well somewhere in this review. For now I think it is simpler to omit it and let the steps face the direction they would anyway. I am however open to follow on work that would for example implement what you suggest.

The complexity of also including rotation is that it's not clear to me that some particular rotation is more correct to be the default than some other. Someone may want slides to have their bottom toward origin, someone may want top toward origin and yet someone may want them the way they are now without doing anything. In such a situation it's usually easiest and least surprising to just do nothing. 

This approach has worked well for the rel plugin as well. The biggest benefit comes from getting x/y/z defined with relative coordinates, and it's fairly straightforward to add rotations as needed. In particular, it would not be easier understand how relative rotations would work.